homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +171,206 @@ def update_state(self, new_state):
+        self.call_service(DOMAIN, SERVICE_VOLUME_MUTE, params)
+
+    def set_volume(self, value):
+        """Send volume step value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set volume to %s', self.entity_id, value)
+
+        self._flag[CHAR_VOLUME] = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_MEDIA_VOLUME_LEVEL: value}
+        self.call_service(DOMAIN, SERVICE_VOLUME_SET, params)
+
+    def set_volume_step(self, value):
+        """Send volume step value if call came from HomeKit."""
+        _LOGGER.debug('%s: Step volume by %s',
+                      self.entity_id, value)
+
+        service = SERVICE_VOLUME_DOWN if value else SERVICE_VOLUME_UP
+        params = {ATTR_ENTITY_ID: self.entity_id}
+        self.call_service(DOMAIN, service, params)
+
+    def set_input_source(self, value):
+        """Send input set value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set active_identifer to %s',
+                      self.entity_id, value)
+
+        source = self.sources[value]
+        self._flag[CHAR_ACTIVE_IDENTIFIER] = True
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_INPUT_SOURCE: source}
+        self.call_service(DOMAIN, SERVICE_SELECT_SOURCE, params)
+
+    def set_remote_key(self, value):
+        """Send remote key value if call came from HomeKit."""
+        _LOGGER.debug('%s: Set remote key to %s', self.entity_id, value)
+
+        if value in MEDIA_PLAYER_KEYS:
+            service = MEDIA_PLAYER_KEYS[value]
+            if service == SERVICE_MEDIA_PLAY_PAUSE:
+                state = self.hass.states.get(self.entity_id).state
+                if state in (STATE_PLAYING, STATE_PAUSED):
+                    self.set_play_pause(state == STATE_PAUSED)
+                    return
+            params = {ATTR_ENTITY_ID: self.entity_id}
+            self.call_service(DOMAIN, service, params)
+
+    def update_state(self, new_state):
+        """Update Television state after state changed."""
+        current_state = new_state.state
+
+        # Power state television
+        hk_state = current_state not in (STATE_OFF, STATE_UNKNOWN)
+        if not self._flag[CHAR_ACTIVE]:
+            hk_state = 1 if hk_state else 0
+            _LOGGER.debug('%s: Set current active state to %s',
+                          self.entity_id, hk_state)
+            self.char_active.set_value(hk_state)
+        self._flag[CHAR_ACTIVE] = False
+
+        if CHAR_VOLUME_SELECTOR in self.chars_speaker:
+            current_mute_state = new_state.attributes.get(
+                ATTR_MEDIA_VOLUME_MUTED)
+            if not self._flag[FEATURE_TOGGLE_MUTE]:
+                _LOGGER.debug('%s: Set current mute state to %s',
+                              self.entity_id, current_mute_state)
+                self.char_mute.set_value(current_mute_state)
+            self._flag[FEATURE_TOGGLE_MUTE] = False
+
+        if self.support_volume_level:
```

Comment:
As indicated earlier, since doesn't have a UI representation in iOS, I think it can be removed.